homeassistant/components/media_player/reproduce_state.py

```diff
@@ -52,45 +56,62 @@ async def call_service(service: str, keys: Iterable) -> None:
         data = {"entity_id": state.entity_id}
         for key in keys:
             if key in state.attributes:
-                data[key] = state.attributes[key]
+                if key == ATTR_MEDIA_POSITION:
+                    # for this property only, the state does not match the service data
+                    data[ATTR_MEDIA_SEEK_POSITION] = state.attributes[key]
```

Comment:
This is incorrect. If the state is playing, you need to take the time the state was created minus `ATTR_MEDIA_POSITION_UPDATED_AT` and add that to the media position.

Here is an example in TypeScript https://github.com/home-assistant/frontend/blob/dev/src/data/media-player.ts#L24

Note that since the state is stored at a different moment, you can't use `utcnow`

Reply:
Good point. It does not look like `scene.create` currently saves this information.

I think the best approach would be to add an attribute to the state as saved in the snapshot, like `ATTR_SNAPSHOT_AT`. This might be useful for other integrations, too.